src/carriers/CMakeLists.txt

```diff
@@ -8,7 +8,7 @@
 include(YarpPlugin)
 include(YarpPrintFeature)
 
-yarp_begin_plugin_library(yarpcar OPTION CREATE_OPTIONAL_CARRIERS
+yarp_begin_plugin_library(yarpcar OPTION YARP_COMPILE_CARRIERS
```

Comment:
You should be adding a line
```cmake
yarp_renamed_option(CREATE_OPTIONAL_CARRIERS YARP_COMPILE_CARRIERS) # since YARP 3.0.0
```
(same for the devices)